easybuild/easyblocks/m/motioncor2.py

```diff
@@ -0,0 +1,114 @@
+##
+# Copyright 2009-2019 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing MotionCor2, implemented as an easyblock
+
+@author: Ake Sandgren, (HPC2N, Umea University)
+"""
+
+import os
+import stat
+
+from easybuild.framework.easyblock import EasyBlock
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import adjust_permissions, copy_file, mkdir, write_file
+from easybuild.tools.modules import get_software_version
+
+
+class EB_MotionCor2(EasyBlock):
+    """
+    Support for installing MotionCor2
+     - creates wrapper that loads the correct version of CUDA before
+     - running the actual binary
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(EB_MotionCor2, self).__init__(*args, **kwargs)
+
+    def extract_step(self):
+        """Extract the files"""
+        super(EB_MotionCor2, self).extract_step()
+
+    def configure_step(self):
+        """No configuration, this is binary software"""
+        pass
+
+    def build_step(self):
+        """No compilation, this is binary software"""
+        pass
+
+    def install_step(self):
+        """
+        Install binary and a wrapper that loads correct CUDA version.
+        """
+
+        cuda_mod_name = ""
+        for dep in self.toolchain.dependencies:
+            if dep['name'] == 'CUDA':
+                cuda_mod_name = dep['short_mod_name']
+                cuda_name = cuda_mod_name.split('/')[0]
+                break
+
+        if cuda_mod_name == "":
+            raise EasyBuildError("CUDA must be a direct dependency of MotionCor2")
+        cuda_ver = get_software_version('CUDA')
```

Comment:
Why not grab the version above in the for loop (`dep['version']`?

We should also move this to the constructor (`__init__`), to ensure `self.motioncor2_bin` is defined when `--module-only` is used... That also means we can't use `get_software_version` (since modules for deps are not loaded yet until `prepare_step` is run, but if you'll use `dep['version']`, that's fine.